### PR TITLE
feat: docs site formatting styling

### PR DESCRIPTION
**Pull request by denniswon**
## PR-Codex overview
This PR focuses on updating the navigation links in the Vitepress config file. 

### Detailed summary
- Updated the link text for the "Getting Started Guide" to "Getting Started"
- Renamed the "Choosing a Smart Account" section to "Smart Accounts"
- Updated the link text for various sections related to smart accounts
- Renamed the "Deployment Addresses" section to "Deployments"
- Updated the link text for various sections related to signers and integrations
- Renamed the "Particle Network Integration Guide" section to "Particle Network"
- Updated the link text for the "Externally Owned Account (EOA)" section to "Using Your Own"
- Renamed the "Overview of aa-core" section to "Getting Started"
- Updated the link text for various sections related to aa-core
- Renamed the "Overview of ISmartContractAccount" section to "Introduction"
- Updated the link text for various sections related to ISmartContractAccount
- Renamed the "Overview of Public Client" section to "Introduction"
- Updated the link text for various sections related to Public Client
- Renamed the "Overview of aa-alchemy" section to "Getting Started"
- Updated the link text for various sections related to aa-alchemy
- Renamed the "Overview of Middleware" section to "Introduction"
- Updated the link text for various sections related to Middleware
- Renamed the "Overview of Utils" section to "Introduction"
- Updated the link text for the "SupportedChains" section
- Renamed the "Overview of aa-accounts" section to "Getting Started"
- Updated the link text for the "LightSmartContractAccount" section to "Light Account"
- Updated the link text for various sections related to LightSmartContractAccount
- Renamed the "Overview of aa-ethers" section to "Getting Started"
- Updated the link text for various sections related to aa-ethers

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: site/.vitepress/config.ts [hunk 1]
```diff
@@ -503,13 +494,8 @@ export default defineConfig({
             text: "Utils",
             collapsed: true,
             base: "/packages/aa-alchemy/utils",
-            items: [
-              {
-                text: "Overview of Utils",
-                link: "/introduction",
-              },
-              { text: "SupportedChains", link: "/supportedChains" },
-            ],
```
Comment (avasisht23) [inline site/.vitepress/config.ts:512]: let's put this back, per my comment re: no section header pages to make each page explicitly visible on sidebar

File: site/.vitepress/config.ts [hunk 2]
```diff
@@ -367,11 +364,8 @@ export default defineConfig({
             text: "Public Client",
             base: "/packages/aa-core/client",
             collapsed: true,
+            link: "/",
             items: [
-              {
-                text: "Overview of Public Client",
-                link: "/",
-              },
```
Comment (avasisht23) [inline site/.vitepress/config.ts:374]: let's put this back, per my comment re: no section header pages to make each page explicitly visible on sidebar

File: site/.vitepress/config.ts [hunk 3]
```diff
@@ -610,11 +596,8 @@ export default defineConfig({
             text: "AccountSigner",
             collapsed: true,
             base: "/packages/aa-ethers/account-signer",
+            link: "/introduction",
             items: [
-              {
-                text: "Overview of AccountSigner",
-                link: "/introduction",
-              },
```
Comment (avasisht23) [inline site/.vitepress/config.ts:617]: let's put this back, per my comment re: no section header pages to make each page explicitly visible on sidebar

File: site/.vitepress/config.ts [hunk 4]
```diff
@@ -641,11 +624,8 @@ export default defineConfig({
             text: "Utils",
             collapsed: true,
             base: "/packages/aa-ethers/utils",
+            link: "/introduction",
             items: [
-              {
-                text: "Overview of Utils",
-                link: "/introduction",
-              },
```
Comment (avasisht23) [inline site/.vitepress/config.ts:648]: let's put this back, per my comment re: no section header pages to make each page explicitly visible on sidebar

File: site/.vitepress/config.ts [hunk 5]
```diff
@@ -519,20 +505,20 @@ export default defineConfig({
         base: "/packages/aa-accounts",
         items: [
           {
-            text: "Overview of aa-accounts",
+            text: "Getting Started",
             link: "/",
           },
           {
-            text: "LightSmartContractAccount",
```
Comment (avasisht23) [inline site/.vitepress/config.ts:526]: I'd keep this name because it's the actual class name. wdyt?

Comment (avasisht23) [inline site/.vitepress/config.ts:526]: I'd keep this name because it's the actual class name. wdyt?

**Review by avasisht23 — CHANGES_REQUESTED** [review]
few more comments

File: site/.vitepress/config.ts (hunk 5 above)
Comment (denniswon) [reply site/.vitepress/config.ts:526]: yeah, but we use just `Light Account` in most of our doc, so thought would be clearer to make the term consistent. I think it's self-explanatory.

**Comment by denniswon** [comment]
@avasisht23 comments accommodated.

**Pull request merged**